jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/DMatrix.java

```diff
@@ -46,12 +49,20 @@ public DMatrix(Iterator<LabeledPoint> iter, String cacheInfo) throws XGBoostErro
     if (iter == null) {
       throw new NullPointerException("iter: null");
     }
+    List<LabeledPoint> points = ImmutableList.copyOf(iter);
```

Comment:
Requiring Guava just for that is probably too much.

Reply:
I'm not sure I like the copying as well. I wonder if we could modify the C++ part to set weights?

Reply:
I have moved the weight setting logic to DataBatch which seems cleaner.